doboto/Endpoint.py

```diff
@@ -80,3 +80,51 @@ def pages(self, request_url, expect, params=None):
                 next_url = None
 
         return items
+
+    def action_result(self, action, wait, poll, timeout):
+        """
+        General action result processor for waiting
+        """
+
+        start_time = time.time()
+
+        while wait and action["status"] == "in-progress":
+
+            time.sleep(poll)
+            try:
+                action = self.do.action.info(action["id"])
+            except:
+                pass
+
+            if time.time() - start_time > timeout:
+                raise DOBOTOException("Timeout on polling", action)
+
+        return action
+
+    def actions_result(self, actions, wait, poll, timeout):
+        """
+        General actions result processor for waiting
+        """
+
+        start_time = time.time()
+
+        info = [index for index, action in enumerate(actions)
+                if action["status"] == "in-progress"]
+
+        while wait and len(info) > 0:
```

Comment:
Ya ya, I see that. Why put it in the loop if it's not really part of the loop.  Plus that list comphrehension was being done if not necessary.